index.js

```diff
@@ -84,6 +84,13 @@ function parse(str, options) {
  */
 
 function serialize(name, val, options) {
+  if (typeof name !== 'string') {
+    throw new TypeError('argument name must be a string');
+  }
+  if (typeof val !== 'string') {
```

Comment:
Would it be enough to skip the verifying that `val` is a string when the `encode` property has been defined for the `options` object? That would allow the custom encoder the user defines to handle `val` however they see fit.